cockroachdb/templates/statefulset.yaml

```diff
@@ -344,16 +365,42 @@ spec:
           resources: {{- toYaml . | nindent 12 }}
         {{- end }}
       volumes:
+      {{- range $i := until (int .Values.conf.store.count) }}
+      {{- if eq $i 0 }}
         - name: datadir
-        {{- if .Values.storage.persistentVolume.enabled }}
+        {{- if $.Values.storage.persistentVolume.enabled }}
           persistentVolumeClaim:
             claimName: datadir
-        {{- else if .Values.storage.hostPath }}
+        {{- else if $.Values.storage.hostPath }}
+          hostPath:
+            path: {{ $.Values.storage.hostPath | quote }}
+        {{- else }}
+          emptyDir: {}
+        {{- end }}
+      {{- else }}
+        - name: datadir-{{ add1 $i }}
+        {{- if $.Values.storage.persistentVolume.enabled }}
+          persistentVolumeClaim:
+            claimName: datadir-{{ add1 $i }}
+        {{- else if $.Values.storage.hostPath }}
           hostPath:
-            path: {{ .Values.storage.hostPath | quote }}
+          path: {{ $.Values.storage.hostPath | quote }}
```

Comment:
hostPath is a directory on a host (node on which cockroachdb pod is running) where a particular volume is mounted. This means the cockroachdb pod writing the data is going on the host at that particular path.

In above example, all the datadir is going on the same hostPath on the same host. I think this will cause issues when multi store is used with hostPath

Reply:
Also `hostPath` indentation on line 387 is also incorrect

Reply:
Fixed. Thank you!